hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/viewfs/ViewFileSystem.java

```diff
@@ -636,15 +630,19 @@ public boolean hasNext() throws IOException {
       @Override
       public LocatedFileStatus next() throws IOException {
         final LocatedFileStatus status = statusIter.next();
-        return (LocatedFileStatus)fixFileStatus(status,
-            getChrootedPath(res, status, f));
+        return (LocatedFileStatus) fixFileStatus(status,
+              getChrootedPath(res, status, f), f, res.isInternalDir());
       }
     };
   }
 
   private Path getChrootedPath(InodeTree.ResolveResult<FileSystem> res,
       FileStatus status, Path f) throws IOException {
     final String suffix;
+    // For internal directories return the qualified path of status
```

Comment:
What is this really doing? It looks like it was already handled as the last case previously.